src/main/java/seedu/algobase/model/searchrule/plansearchrule/TimeRange.java

```diff
@@ -0,0 +1,61 @@
+package seedu.algobase.model.searchrule.plansearchrule;
+
+import static seedu.algobase.commons.util.AppUtil.checkArgument;
+import static seedu.algobase.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.algobase.logic.parser.ParserUtil.FORMATTER;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a {@code TimeRangePredicate}'s starting date and end date.
+ */
+public class TimeRange {
+
+    public static final String MESSAGE_CONSTRAINTS = "Both start date and end date should be present.";
+
+    public final LocalDate startDate;
+    public final LocalDate endDate;
+
+    /**
+     * Constructs a {@code TimeRange}.
+     *
+     * @param startDate starting date of time range
+     * @param endDate end date of time range
+     */
+    public TimeRange(LocalDate startDate, LocalDate endDate) {
+        requireAllNonNull(startDate, endDate);
+        checkArgument(isValidRange(startDate, endDate), MESSAGE_CONSTRAINTS);
```

Comment:
`isValidRange` checks if both are not null, but this is already done by `requireAllNonNull`